### PR TITLE
[rn] Reland support loading model from buffer for Android

### PR DESCRIPTION
**Pull request by YUNQIUGUO**
### Description

Reland previous reverted changes for loading model from buffer - Android 

### Motivation and Context

#13903

### CODE REVIEW

**Comment by YUNQIUGUO** [comment]
Seems the latest React Native CI Run is green now: https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=976445&view=logs&s=28fb2bf2-39c5-5feb-1887-4904233f6193

We need to document somewhere that when touching implementation in js/react_native/android project, always update the index.android.bundle file. and fyi:

`index.android.bundle` file is usually included in the app's assets directory and should being loaded correctly during runtime (In release mode only). We are now running detox e2e tests with release config to avoid a react-native metro server starting on the side, so in this release mode, the e2e app will need to load the pre-bundled JavaScript code from this file instead of from the development server.

File: js/react_native/android/src/main/java/ai/onnxruntime/reactnative/OnnxruntimeModule.java [hunk 1]
```diff
@@ -76,10 +85,30 @@ public void loadModel(String uri, ReadableMap options, Promise promise) {
...
+   *       when run() is called, the key must be passed into the first parameter.
+   */
+  @ReactMethod
+  public void loadModelFromBase64EncodedBuffer(String data, ReadableMap options, Promise promise) {
+    try {
+      byte[] modelData = Base64.decode(data, Base64.DEFAULT);
+      WritableMap resultMap = loadModel(modelData, options);
+      promise.resolve(resultMap);
+    } catch (Exception e) {
+      promise.reject("Can't load model from buffer: " + e.getMessage(), e);
```
Comment (skottmckay) [inline js/react_native/android/src/main/java/ai/onnxruntime/reactnative/OnnxruntimeModule.java:104]:
```suggestion
      promise.reject("Failed load model from buffer: " + e.getMessage(), e);
```
nit: "can't" makes it sound like we don't support loading from buffer.

File: js/react_native/lib/binding.ts [hunk 2]
```diff
@@ -64,6 +64,7 @@ export declare namespace Binding {
 
   interface InferenceSession {
     loadModel(modelPath: string, options: SessionOptions): Promise<ModelLoadInfoType>;
+    loadModelFromBase64EncodedBuffer?(buffer: string, options: SessionOptions): Promise<ModelLoadInfoType>;
```
Comment (skottmckay) [inline js/react_native/lib/binding.ts:67]: Do we have any react native tests that can be updated to call this?

Comment (skottmckay) [reply js/react_native/lib/binding.ts:67]: As discussed. Changes were manually verified in an expo app. Need to update our e2e RN tests in the next week or two but want this change to make the release.

**Review by skottmckay — APPROVED** [review]

**Pull request merged**